db/blob/blob_file_completion_callback.h

```diff
@@ -0,0 +1,52 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+#include "db/error_handler.h"
+#include "file/sst_file_manager_impl.h"
+#include "rocksdb/options.h"
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class BlobFileCompletionCallback {
+ public:
+  BlobFileCompletionCallback(ImmutableDBOptions immutable_db_options,
+                             InstrumentedMutex* mutex,
+                             ErrorHandler* error_handler)
+      : immutable_db_options_(immutable_db_options),
+        mutex_(mutex),
+        error_handler_(error_handler) {}
+
+  Status Callback(const std::string& file_name) {
+    Status s;
+#ifndef ROCKSDB_LITE
+    auto sfm = static_cast<SstFileManagerImpl*>(
+        immutable_db_options_.sst_file_manager.get());
```

Comment:
I think we don't need to `static_cast` here: both `OnAddFile` and `IsMaxAllowedSpaceReached` are part of the `SstFileManager` interface. If so, it would also suffice to `#include "rocksdb/sst_file_manager.h"` instead of the private header containing the `Impl` class.